### ISSUE TITLE
Space setting category doesn't show up

### ISSUE THREAD

**Issue by restitux** [vector-im/element-android#7989]
The space setting category doesn't show up. This blocks users from changing settings like "Show all rooms in home space". New app users cannot change this setting at all, forcing them into space filters when interacting with notifications for rooms in spaces (similar to the issue fixed in https://github.com/vector-im/element-android/pull/5827).

Here is where the visibility settings for the spaces category are defined:
https://github.com/vector-im/element-android/blob/6b98b3023e6507b211d849d14dcb167c130cd3d4/vector/src/main/java/im/vector/app/features/settings/VectorSettingsPreferencesFragment.kt#L105-L108

Here is where `isNewAppLayoutFeatureEnabled` is hardcoded to true. This causes `prefs.isVisible` to always be false.
https://github.com/vector-im/element-android/blob/6b98b3023e6507b211d849d14dcb167c130cd3d4/vector/src/main/java/im/vector/app/features/VectorFeatures.kt#L62

**Comment by mnaturel** [comment]
@restitux Thanks for raising the issue. Not sure but it seems to me we should use `vectorPreferences.isNewAppLayoutEnabled()` instead of `vectorFeatures.isNewAppLayoutFeatureEnabled() ` for the visibility property. I have seen it also needs to be updated in other part of the code.

**Comment by criticalAY** [comment]
Hey! I took care of it

**Comment by restitux** [comment]
Looking at @criticalAY 's PR, is there any reason why the spaces settings are gated behind the new layout being enabled? The "Show all spaces in Home space" setting seems applicable to both layouts supported by Element Android, especially since it also controls whether filters are automatically applied when launching the app via message notifications.

**Comment by wapsi** [comment]
I had to reset my Element on my Android tablet (clearing cache & data), and after I started it again, it's missing the setting "Show all rooms in home space". Is it possible to toggle that on somehow?

**Comment by wapsi** [comment]
When the "Enable new layout" is disabled in the "Labs" menu, there should be "Show all rooms in home space" shown in Preferences section:

![image](https://user-images.githubusercontent.com/11405096/217493444-8dc66d94-ae0e-4981-bb3f-9e1899561b5f.png)

**Comment by restitux** [comment]
I'm confused why this settings panel is limited to one of the layouts. The setting in question (Show all rooms in Home) applies in both layouts AFAIK.

**Comment by wapsi** [comment]
I agree, can we then just remove those conditions mentioned in the first message (and display the setting always)?

**Comment by mnaturel** [comment]
Hello, so to clarify the situation about this setting:
- in the new app layout, it is no more relevant since we now have a dedicated "All chats" entry in the space list filter by pressing the space list button, the entry "Home" no longer exists
- in the old app layout, it is relevant. So the setting should be visible when the labs setting for new layout is disabled and not when the feature is disabled. Let me fix this.

**Comment by wapsi** [comment]
I installed the 1.5.24 version on my Android tablet, but the setting is not still there, even in the changelog there is:
> Space setting category doesn't show up by @mnaturel in https://github.com/vector-im/element-android/pull/8160

I also tried to enable and then disable the new layout / view, but the setting is still not showing up. Is there need to clear the cache and data, and re-login into Matrix?

**Comment by restitux** [comment]
@mnaturel the problem with only showing it for the old layout is that it breaks the fix in https://github.com/vector-im/element-android/pull/5827. That change made it so you have to enable this setting to disable the auto space focus on notification click behavior. If this setting is not available in the new layout, you are unable to switch this behavior without going back to the old layout.

**Comment by wapsi** [comment]
Any news about this? I think this issue should be reopened, because it's not possible to set that "Show all rooms in home" setting on, because it's still missing from the view if the new layout is disabled...?

**Comment by wapsi** [comment]
As a workaround I edited manually the file: /data/data/im.vector.app/shared_prefs/im.vector.app_preferences.xml file, by chaning the parameter:
`<boolean name="SETTINGS_PREF_SPACE_SHOW_ALL_ROOM_IN_HOME" value="false" />`
to:
`<boolean name="SETTINGS_PREF_SPACE_SHOW_ALL_ROOM_IN_HOME" value="true" />`

So I think the setting and functionality is definitely there, but this setting is only missing from the UI, even the "new layout" is not activated...

**Comment by wapsi** [comment]
@restitux / @mnaturel: Could you please reopen this issue, or should I create a new issue and refer to this?

There is definitely bug in the Android Element now, because it does not show the Space category setting at all in the UI, even it's possible manually to tune that value in its XML configuration file. And this used to work just fine earlier (the setting was visible in the UI), but I think the new layout and its changes messed it up.

**Comment by mnaturel** [comment]
Hello @bmarty , maybe you could have a look to this issue ? I will let you check what should be done with it.
